src/components/sidebar/SafeListItem/index.tsx

```diff
@@ -39,7 +39,7 @@ const SafeListItem = ({
   const chain = useAppSelector((state) => selectChainById(state, chainId))
   const allAddressBooks = useAppSelector(selectAllAddressBooks)
   const currChainId = useChainId()
-  const isCurrentSafe = currChainId === currChainId && safeAddress.toLowerCase() === address.toLowerCase()
+  const isCurrentSafe = chainId === currChainId && sameAddress(safeAddress, address)
   const name = allAddressBooks[chainId]?.[address]
   const isOpen = sameAddress(address, safeAddress)
```

Comment:
Unrelated but I noticed a small bug on this line if there are safes with the same address on different chains, both will be highlighted. Extending this flag by `chainId === currChainId` would fix that as well.